lib/java_buildpack/component/java_opts.rb

```diff
@@ -42,6 +42,16 @@ def add_javaagent(path)
         add_preformatted_options "-javaagent:#{qualify_path path}"
       end
 
+      # Adds a +agentpath+ entry to the +JAVA_OPTS+.  Prepends +$PWD+ to the path (relative to the droplet root) to
+      # ensure that the path is always accurate.
+      #
+      # @param [Pathname] path the path to the +agentpath+ shared library
+      # @param [Properties] properties to append to the agentpath entry
+      # @return [JavaOpts]     +self+ for chaining
+      def add_agentpath_with_props(path, props)
+        add_preformatted_options "-agentpath:#{qualify_path path}=" + props.map { |k, v| "#{k}=#{v}" }.join(',')
+      end
+
```

Comment:
Could you add an explicit test for this method to https://github.com/cloudfoundry/java-buildpack/blob/master/spec/java_buildpack/component/java_opts_spec.rb

I have just noticed that the plain `add_agentpath` has no explicit test but is still covered from other tests. It's a simple method so I happy to add tests for this `with_props` one at the same time once this pull request is in.